Build failure: mgmt

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

nix-build -A mgmt

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/309043726

### Relevant log output

```console
patching script interpreter paths in misc/header.sh
misc/header.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xhsnsmvh1ka5mxszd1psxq36vaanb8jy-bash-5.3p3/bin/bash"
bash: line 1: git: command not found
bash: line 1: git: command not found
Generating: lexer...
Generating: parser...
Generating: interpolation...
Generating: type kind strings...
Generating: funcs...
Building subPackage ./.
# github.com/purpleidea/mgmt/lang/funcs/core
lang/funcs/core/generated_funcs.go:2060:20: undefined: runtime.Bool
lang/funcs/core/generated_funcs.go:2075:20: undefined: runtime.Bytes
lang/funcs/core/generated_funcs.go:2089:20: undefined: runtime.Float64
lang/funcs/core/generated_funcs.go:2102:20: undefined: runtime.Int64
lang/funcs/core/generated_funcs.go:2116:20: undefined: runtime.String
lang/funcs/core/generated_funcs.go:2353:14: undefined: strings.HasPrefixFold
```

### Additional context

This package is likely out of date. Should we remove or fix?

### System metadata

n/a

### Notify maintainers


@urandom2 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -13283,6 +13283,13 @@
     githubId = 34152449;
     name = "Karl Hallsby";
   };
+  karpfediem = {
+    name = "Karpfen";
+    github = "karpfediem";
+    githubId = 11753414;
+    email = "nixos@karpfen.dev";
+    matrix = "@karpfen:matrix.org";
+  };
   kashw2 = {
     email = "supra4keanu@hotmail.com";
     github = "kashw2";

--- a/pkgs/by-name/mg/mgmt/package.nix
+++ b/pkgs/by-name/mg/mgmt/package.nix
@@ -52,11 +52,14 @@ buildGoModule (finalAttrs: {
 
   subPackages = [ "." ];
 
-  meta = with lib; {
+  meta = {
     description = "Next generation distributed, event-driven, parallel config management";
     homepage = "https://mgmtconfig.com";
     license = lib.licenses.gpl3Only;
-    maintainers = with lib.maintainers; [ urandom ];
+    maintainers = with lib.maintainers; [
+      urandom
+      karpfediem
+    ];
     mainProgram = "mgmt";
   };
 })

--- a/pkgs/by-name/mg/mgmt/package.nix
+++ b/pkgs/by-name/mg/mgmt/package.nix
@@ -10,28 +10,24 @@
   pkg-config,
   ragel,
 }:
-buildGoModule rec {
+buildGoModule (finalAttrs: {
   pname = "mgmt";
-  version = "unstable-2022-10-24";
+  version = "1.0.1";
 
   src = fetchFromGitHub {
     owner = "purpleidea";
     repo = "mgmt";
-    rev = "d8820fa1855668d9e0f7a7829d9dd0d122b2c5a9";
-    hash = "sha256-jurZvEtiaTjWeDkmCJDIFlTzR5EVglfoDxkFgOilo8s=";
+    tag = finalAttrs.version;
+    hash = "sha256-Qi9KkWzFOqmUp5CSHxzQabQ8bVnBbxxKS/W6aLBTv6k=";
   };
 
-  # patching must be done in prebuild, so it is shared with goModules
-  # see https://github.com/NixOS/nixpkgs/issues/208036
-  preBuild = ''
-    for file in `find -name Makefile -type f`; do
-      substituteInPlace $file --replace "/usr/bin/env " ""
-    done
+  vendorHash = "sha256-XZTDqN5nQqze41Y/jOhT3mFHXeR2oPjXpz7CJuPOi8k=";
 
-    substituteInPlace lang/types/Makefile \
-      --replace "unset GOCACHE && " ""
+  postPatch = ''
     patchShebangs misc/header.sh
-    make lang funcgen
+  '';
+  preBuild = ''
+    make lang resources funcgen
   '';
 
   buildInputs = [
@@ -50,19 +46,17 @@ buildGoModule rec {
   ldflags = [
     "-s"
     "-w"
-    "-X main.program=mgmt"
-    "-X main.version=${version}"
+    "-X main.program=${finalAttrs.pname}"
+    "-X main.version=${finalAttrs.version}"
   ];
 
   subPackages = [ "." ];
 
-  vendorHash = "sha256-Dtqy4TILN+7JXiHKHDdjzRTsT8jZYG5sPudxhd8znXY=";
-
   meta = with lib; {
     description = "Next generation distributed, event-driven, parallel config management";
     homepage = "https://mgmtconfig.com";
-    license = licenses.gpl3Only;
-    maintainers = with maintainers; [ urandom ];
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ urandom ];
     mainProgram = "mgmt";
   };
-}
+})

